src/hooks/use-current-page.ts

```diff
@@ -12,17 +13,27 @@ const cidRegex = /(?<=\/)[^/?]+(?=\?|$)/
 
 const filenameRegex = /(?<=filename=)[^&]+/
 
+const maddrsRegex = /(?<=maddrs=)[^&]+/
+
 export type CurrentPage = 'add' | 'download'
 export const useCurrentPage = (): CurrentPage => {
   const [location] = useHashLocation()
-  const { setDownloadInfo } = useDownloadInfo()
+  const dispatch = useFilesDispatch()
   const maybeCid = location.match(cidRegex)?.[0] ?? null
   const filename = location.match(filenameRegex)?.[0] ?? null
+  const maddrs = location.match(maddrsRegex)?.[0] ?? null
 
+  // Dispatch the fetch_start action if the URL contains a cid
   useEffect(() => {
     if (maybeCid == null) return
-    setDownloadInfo(maybeCid, filename)
-  }, [maybeCid, filename])
+    dispatch({ type: 'reset_files' })
+    const decodedFilename = decodeURIComponent(filename ?? '')
+    // Decode the provider's maddrs from the URL
+    const decodedMaddrs = maddrs != null ? decodeURIComponent(maddrs).split(',') : null
+    const multiaddrs = decodedMaddrs != null ? decodedMaddrs.map(maddr => multiaddr(maddr)) : null
```

Comment:
nit: I feel like a separate `useMemo` on a `maddrs` dep with the fully resolved `multiaddrs` would be a little better.. but there is likely little difference.. sometimes react can get wonky if you have different effects too separated.